supervision/annotators/core.py

```diff
@@ -976,94 +1169,53 @@ def __init__(
         color_lookup: ColorLookup = ColorLookup.CLASS,
         border_radius: int = 0,
         smart_position: bool = False,
+        ensure_in_frame: bool = False,  # Inherited
     ):
-        """
-        Args:
-            color (Union[Color, ColorPalette]): The color or color palette to use for
-                annotating the text background.
-            text_color (Union[Color, ColorPalette]): The color or color palette to use
-                for the text.
-            text_scale (float): Font scale for the text.
-            text_thickness (int): Thickness of the text characters.
-            text_padding (int): Padding around the text within its background box.
-            text_position (Position): Position of the text relative to the detection.
-                Possible values are defined in the `Position` enum.
-            color_lookup (ColorLookup): Strategy for mapping colors to annotations.
-                Options are `INDEX`, `CLASS`, `TRACK`.
-            border_radius (int): The radius to apply round edges. If the selected
-                value is higher than the lower dimension, width or height, is clipped.
-            smart_position (bool): Spread out the labels to avoid overlapping.
-        """
-        self.border_radius: int = border_radius
-        self.color: Union[Color, ColorPalette] = color
-        self.text_color: Union[Color, ColorPalette] = text_color
         self.text_scale: float = text_scale
         self.text_thickness: int = text_thickness
-        self.text_padding: int = text_padding
-        self.text_anchor: Position = text_position
-        self.color_lookup: ColorLookup = color_lookup
-        self.smart_position = smart_position
+        super().__init__(
+            color=color,
+            text_color=text_color,
+            text_padding=text_padding,
+            text_position=text_position,
+            color_lookup=color_lookup,
+            border_radius=border_radius,
+            smart_position=smart_position,
+            ensure_in_frame=ensure_in_frame,
+        )
 
     @ensure_cv2_image_for_annotation
     def annotate(
         self,
-        scene: ImageType,
+        scene: np.ndarray,  # Ensure scene is initially a NumPy array here
```

Comment:
```suggestion
        scene: ImageType,  # Ensure scene is initially a NumPy array here
```

Please use ImageType we want to support both Pillow and Opencv (numpy arrays)